src/System.Private.ServiceModel/src/System/ServiceModel/Channels/LayeredChannel.cs

```diff
@@ -57,12 +58,12 @@ protected override void OnClose(TimeSpan timeout)
 
         protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
         {
-            return _innerChannel.BeginClose(timeout, callback, state);
+            throw ExceptionHelper.PlatformNotSupported();
```

Comment:
>throw ExceptionHelper.PlatformNotSupported(); [](start = 12, length = 45)

Reviewers: PNSE's like this are a deliberate attempt to make all our CommunicationObjects enforce the fact we have moved to the OnOpenAsync and OnCloseAsync model across the board.  It is not strictly related to WS* support other than continuing our existing practice in newly added/modified types. It should not be possible for an app to cause these methods to be called, regardless if they use sync or async open.   But they are useful to uncover coding mistakes of our own and to discover dead AsyncResult code.